### ISSUE TITLE
Gradle can't find arm64 gcc/clang

### ISSUE THREAD

**Issue by ryanmkurtz** [gradle/gradle#17660]
I am trying to get [Ghidra](https://github.com/NationalSecurityAgency/ghidra) to build on `arm64` Linux and macOS and had to fiddle with the `toolChains` in order to get it to work on those platforms.  Ultimately the minimal thing I needed to add to get it to work on these (and all other) platforms was:

```
	// The toolChains block is needed because Gradle doesn't find gcc/clang on ARM-based Linux
	// and macOS platforms without explicitly declaring their targets.  Might be a bug in the 
	// native Gradle plugin. 
	toolChains {
		if (isCurrentLinux()) {
			gcc(Gcc) {
				if (isCurrentArm_64()) {
					target("linux_arm_64")
				}
			}
		}
		if (isCurrentMac()) {
			clang(Clang) {
				if (isCurrentArm_64()) {
					target("mac_arm_64")
				}
			}
		}
	}
```
The `isCurrentArm_64()` function is just checking the JVM's `os.arch` property, looking for different types of arm64 identifiers (arm64, aarch64, etc.)

If I leave out the OS checks then I think it was trying to use clang on Linux even though gcc comes before it in the list...I was getting errors about xcrun.  That was a bit unexpected too.

### Expected Behavior
We didn't need a `toolChains` block before when we were just building on `x86_64` for Windows, Linux, and macOS.  It was just able to find visual studio, gcc, and clang on the path.  I would expect it to correctly identify that it's on a platform with `arm64` architecture and still use those same standard compilers found on the `PATH`.

### Your Environment
Gradle 7.1.1
Java 11.0.10 (Amazon Corretto)
Ubuntu 20.04.1 aarch64, macOS M1, Windows 10 x86_64

**Comment by ghidra1** [comment]
In defining `defaultPlatformDefinitions` the class `org.gradle.nativeplatform.platform.internal.NativePlatforms` attempts lookup of `armv8` architecture which will fail and return a `DefaultArchitecture` without the other aliases. The `org.gradle.nativeplatform.platform.internal.Architectures` class should update its 64-bit ARM `KnownArchitecture` construction to include `armv8`:
`
new KnownArchitecture("arm-v8", "armv8", "arm64"),
`

**Comment by ghidra1** [comment]
It would also seem appropriate to change `org.gradle.nativeplatform.platform.internal.ArchitectureInternal` to add an `isArm64()` method and avoid the ambiguity with the `isArm()` method.

**Comment by ryanmkurtz** [comment]
Bump

**Comment by PriNova** [comment]
Need it too for aarch64 linux on android

**Comment by sgammon** [comment]
Bump

**Comment by ryanmkurtz** [comment]
Possibly fixed by e47459f9a99064b832b4a9f1237018672bd4ce29.  Need to test.

**Comment by ryanmkurtz** [comment]
I can confirm that this issue is fixed in 8.11.

I suspect a similar fix will be needed for FreeBSD though.
